### DISCUSSION TITLE
How to Train 2 Subject People with the New Method?

### DISCUSSION THREAD

**Discussion by MasterDenis** [TheLastBen/fast-stable-diffusion#278] [Q&A]
Hello there wonderful community!
So lately I was testing the new method and was thinking of adding two named people into the training for parallel training in order to have them presented in the same composition later for the best results, for now I work with a colab notebook so I would like to ask which is the best way to go about it with the lowest ram available for colab,

I was wondering if by naming the subjects differently on their photos that would help or if there should be another approach.
i.e. Jimmy (1), Jimmy (2)... and Marie (1), Marie (2)...

**Comment by TheLastBen** [comment]
Simply by naming the pictures with unique identifiers will let you train on multiple subjects, keep the number of instance images as low as 15 per person.

And most importantly, you shouldn't use regular names for the instances, instead of jimmy, use a random short password-like name like : "jjjmmkttvbv" and make sure it doesn't contain known abbreviations. Otherwise the training won't be as efficient.

**Reply by TheLastBen** [reply]
jjjmmkttvbv (1).jpg .... jjjmmkttvbv (15).jpg
mrrridfdree (1).jpg ..... mrrridfdree (15).jpg

If the total number of instance images = 15, the total steps should be around 2000. So for 2 subjects, do 4000 steps.

You can resume training as many times as you want if you're not satisfied with the result

**Reply by MasterDenis** [reply]
Hello Again, after a very long time of testing I have come to specific results, by using at first about 140 total images I followed the advice of images * 100 steps and ran it for about 14k steps, this caused any prompt to result in a very messy and distorted generic picture of the subjects both in the same frame when summoned at the same time, that is one issue that I encountered.

On another pass, I tested of course with the suggested 15 images per person, for a total of about 3000 steps (30 images 3k steps), this also did not produce satisfactory results because one of the subjects would always come up a bit blurry/messy on the face, just to clarify, though... how do you calculate the average of how many steps you need using the new fast method?

also I was wondering if you have inside that method a preservation mode for the model (similar to other notebooks where the preservation is on by default to avoid any issues).

Thanks for the answers so far!

**Reply by TheLastBen** [reply]
From my experience, I'm getting acceptable results with 300 steps, but I'm using unknown identifiers and good quality instance images : 
I added a feature for text_encoder limiting, use the latest notebook to get it and give it a try, try a celebrity this time so we can do a comparaison

example of 300 steps : 

![80 + 300](https://user-images.githubusercontent.com/110583491/198905026-51470f9c-360d-4fcb-9e76-4c86eb395405.png)

**Reply by MasterDenis** [reply]
Yeah, hold on, you suggest about x100 steps to the number of images so images*100, did you use 3 images in that case?
In such casehat should the ideal be for someone with 15-20 instance images lets say for a human face with a few zoomed-out photos for variety?
for people, I tried a few, someone suggested x83 (but on Shiv's Notebook, no idea how this would apply on yours).

**Reply by TheLastBen** [reply]
I actually used 29 images, the important thing is good quality input images

**Reply by MasterDenis** [reply]
- Also could you add to the notebook what exactly the new feature does? Because I for example as graphic designer and tester do not know exactly what it does on the training session. (there is also a typo there, check)

- as for one person with good quality images (different background, same person on slightly different poses) would you say images*83 is a good way to go for steps?

**Reply by TheLastBen** [reply]
keep the number of the instance images below 30, and the steps 30*100 if your using 100% text_encoder, 30*150 or more if you're using 10% text_encoder

**Reply by MasterDenis** [reply]
do you mean 30100 steps or is that wrong?

**Reply by TheLastBen** [reply]
start by training 1person and see the result first

**Reply by giteeeeee** [reply]
> keep the number of the instance images below 30, and the steps 30_100 if your using 100% text_encoder, 30_150 or more if you're using 10% text_encoder

Curious about why such drastic leap in %? Should I also try maybe 30% 50% 70%?

**Reply by TheLastBen** [reply]
Higher value puts more weight on the instance, meaning overfitting and harder to stylize

**Reply by giteeeeee** [reply]
> Higher value puts more weight on the instance, meaning overfitting and harder to stylize

but why recommending 10%? If its effect is additive, isn't 10% almost like turning the encoder off?

**Reply by TheLastBen** [reply]
the text_encoder replaces the instance prompt, once it determines what type of subject you're training, it would be enough, if you keep training the text_encoder after that, it will make the instance harder to stylize, 10% is the minimum, you can go up to 50% and be able to stylize

**Reply by giteeeeee** [reply]
> the text_encoder replaces the instance prompt, once it determines what type of subject you're training, it would be enough, if you keep training the text_encoder after that, it will make the instance harder to stylize, 10% is the minimum, you can go up to 50% and be able to stylize

Interesting. So the use of instance prompt + text encoder together, resulted in overfitting with the old method...
Thanks for the explanation, always great help

**Reply by MasterDenis** [reply]
> the text_encoder replaces the instance prompt, once it determines what type of subject you're training, it would be enough, if you keep training the text_encoder after that, it will make the instance harder to stylize, 10% is the minimum, you can go up to 50% and be able to stylize

can you make  on the end of the notebook a section where all those terms are explained in detail for best use? That would help to clear some confusion, for instance I did not know what the text encoder does prior to your explanation, if it is written down in the end of the notebook people can ready it without adding too much clutter on the executable code.

**Reply by TheLastBen** [reply]
Check the latest update, I explained the effect of the text_encoder on the training : https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/fast-DreamBooth.ipynb

**Comment by EricBintner** [comment]
This is great info. I also got a little confused on the new version. I don’t see a place to define a class. Last time I trained a model I specifically defined “dog” for the class, but now I’m not sure how to do that. I ran it again yesterday and got a successful set, but I never defined a class (in this case “person”) and SD runs fine without defining the class in the prompt.
